src/Ontology/Address.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Hash.h"
+#include "Address.h"
+#include "HexCoding.h"
+
+#include <stdexcept>
+
+#include <TrezorCrypto/sha2.h>
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ripemd160.h>
+
+using namespace TW::Ontology;
+
+Address::Address(const PublicKey &publicKey) {
+    std::vector<uint8_t> builder(publicKey.bytes);
+    builder.insert(builder.begin(), 0x21);
+    builder.push_back(0xAC);
```

Comment:
Let's define constants for these magic numbers.

Reply:
This is VM op code, we only need two op code. Besides, the op code's name make no sense in here. You will see more op code in transaction serialization, we will also provide op code in this way. otherwise, I will spend more than a month to handle transaction serialization. It's unacceptable.

Reply:
I add op code's name in this commit: https://github.com/TrustWallet/wallet-core/pull/103/commits/8eac3fbeb8a318e3a7139983fe6012f9ec3bc28f